mani_skill/vector/wrappers/gymnasium.py

```diff
@@ -141,12 +142,13 @@ def step(
         dones = torch.logical_or(terminations, truncations)
 
         if dones.any() and self.auto_reset:
-            final_obs = obs
+            final_obs = copy.deepcopy(obs)
             env_idx = torch.arange(0, self.num_envs, device=self.device)[dones]
-            obs, _ = self.reset(options=dict(env_idx=env_idx))
-            infos["final_info"] = infos.copy()
+            final_info = copy.deepcopy(infos)
+            obs, infos = self.reset(options=dict(env_idx=env_idx))
```

Comment:
Is gymnasium infos supposed to be done like this?

If i recall correctly generally on the first environment reset there is no useful infos, just reset info details. Althoughin the ManiSkill case the reset info is using information from the get_info function.

Reply:
I am assuming that the existing `self.reset` does what we want, since this `obs, infos = self.reset(options=dict(env_idx=env_idx))` is called inside `step` and so the env will already have performed its initial `reset`, do you think this is correct for maniskill?